### DISCUSSION TITLE
How to fix WandB plotting issues?

### DISCUSSION THREAD

**Discussion by djberenberg** [Lightning-AI/pytorch-lightning#8390] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I'm having some trouble with a few of the visualizations available through the `WandBLogger`, I'm wondering if anyone has experienced similar issues.

The steps to reproduce this involve something like:
 
```python3
# previously logged in via `wandb login`
pl_module = SomeLightningModule()
wandb_logger = pl.loggers.WandBLogger()
wandb_logger.watch(pl_module.component1, 'gradients' , log_freq=1)
wandb_logger.watch(pl_module.component2, 'gradients' , log_freq=1) # could be a different log_freq
...
trainer = pl.Trainer(max_epochs=10, ...)
train, val = GET_DATA(...)
trainer.fit(pl_module, train, val)
```
1. The values in the metrics plots do not match the values reported to `stdout`.
Below I am showing one example.
Here is the WandB visualization, note the loss for epoch 5:
<img width="378" alt="Screen Shot 2021-07-12 at 8 28 12 PM" src="https://user-images.githubusercontent.com/22937224/125371992-007ccd80-e350-11eb-8557-3ae7198f6f47.png">
and the terminal output for epoch 5:
<img width="511" alt="image" src="https://user-images.githubusercontent.com/22937224/125372092-3d48c480-e350-11eb-8c9d-23659d597cb0.png">

2. For `log_freq=1`, the gradient visualizations collapse. Screenshot seen below. Could this have to do with the fact that no gradients are reported during validation steps?
<img width="378" alt="image" src="https://user-images.githubusercontent.com/22937224/125372254-a3cde280-e350-11eb-9d88-2511cfc33b3f.png">

Other info:
- I am using multiple dataloaders under the `min_size` iteration paradigm, one (validation) dataset is much longer than the other. 
- At the end of my training/validation steps I call `self.log_dict` in my `LightningModule` to log/aggregate metrics
